src/pages/solve/MaxCut.tsx

```diff
@@ -0,0 +1,58 @@
+import { Center, Divider, Text, Textarea } from "@chakra-ui/react";
+import { NextPage } from "next";
+import Head from "next/head";
+import React, { useState } from "react";
+import { Container } from "../../components/Container";
+import { Main } from "../../components/Main";
+import { GraphArea } from "../../components/solvers/Graph/GraphArea";
+import { ProgressHandler } from "../../components/solvers/ProgressHandler";
+import { SolverTitle } from "../../components/solvers/SolverTitle";
+import { parseGML } from '../../converter/graph/gml/GmlParser';
+
+const MaxCut: NextPage = () => {
+    const [graphData, setGraphData] = useState<any>( null);
+
+    const [graphString, setGraphString] = useState("");
+    const [errorString, setErrorString] = useState("");
+
+    return (
+        <Container minHeight="100vh">
+            <Head>
+                <title>ProvideQ</title>
+                <meta name="description" content="Generated by create next app" />
+                <link rel="icon" href="/favicon.ico" />
+                {/* TODO: replace favicon */}
+            </Head>
+            <SolverTitle title="MaxCut Solver" text="For a given undirected, weighted graph, this algorithm finds a cut that is a maximum in some way or another." />
+            <Main mb="20vh">
+
+                <Textarea placeholder="Enter your graph in GML format"
+                          value={graphString}
+                          minHeight="10rem"
+                          isInvalid={errorString != ""}
+                          onChange={e => {
+                    setGraphString(e.target.value);
+
+                    try {
+                        let data = parseGML(e.target.value);
+                        setGraphData(data);
+
+                        setErrorString('');
+                    } catch (e: any) {
+                        setErrorString(e.message);
+                    }
+                }} />
```

Comment:
code style: please move this function out of the JSX part for better readability

Reply:
moved to its own function in [a83498a](https://github.com/ProvideQ/ProvideQ/pull/4/commits/a83498a1f227b2939eee3514d61b39eeca3be5ac)